x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,346 @@
+          event_types: schema.arrayOf(schema.string()),
+          file_path: schema.maybe(schema.string()),
+          signer: schema.maybe(schema.string()),
+          sha256: schema.maybe(schema.string()),
+          dismiss: schema.boolean({ defaultValue: false }), // Boolean determining if we dismiss all alerts that match this
+        }),
+      },
+      options: { authRequired: true },
+    },
+    handleWhitelistPost
+  );
+
+  router.delete(
+    {
+      path: allowlistBaseRoute,
+      validate: {
+        body: schema.object({
+          whitelist_id: schema.string(),
+        }),
+      },
+      options: { authRequired: true },
+    },
+    handleWhitelistItemDeletion
+  );
+}
+
+/**
+ * Handles the POST request for whitelist additions
+ */
+async function handleWhitelistPost(context: RequestHandlerContext, req, res) {
+  try {
+    const eventTypes: string[] = req.body.eventTypes;
+    const whitelistAttributeMap: Record<string, string> = {
+      file_path: 'malware.file.path',
+      acting_process_path: 'actingProcess.file.path',
+      signer: 'malware.file.signer',
+      sha256: 'malware.file.hashes.sha256',
+    };
+    const newRules: WhitelistRule[] = [];
+    Object.keys(whitelistAttributeMap).forEach(k => {
+      if (req.body[k]) {
+        newRules.push({
+          comment: req.body.comment || '',
+          event_types: eventTypes,
+          entry_type: 'simple',
+          entry: {
+            comparison: 'equality',
+            value: req.body[k],
+            apply_to: whitelistAttributeMap[k],
+          },
+        });
+      }
+    });
+
+    // Don't index an empty list if no rules could be created from the request
+    if (newRules.length === 0) {
+      return res.badRequest({
+        error: i18n.translate('allowlist.badrequest', {
+          defaultMessage: 'No allowlist rules could be created from request.',
+        }),
+      });
+    }
+
+    // Check that the global whitelist size limit won't be reached
+    if (newRules.length + whitelistCount > whitelistLimit) {
+      return res.badRequest({
+        error: i18n.translate('allowlist.sizelimit.exceeded', {
+          defaultMessage: 'Allowlist limit of {whitelistLimit} reached.',
+          values: { whitelistLimit },
+          description: 'Allowlist limit exceeded.',
+        }),
+      });
+    }
+
+    // Add the rules to the global whitelist
+    const createdItemIDs = await addWhitelistRule(context, newRules);
+    if (createdItemIDs.length === 0) {
+      return res.internalError({
+        error: i18n.translate('allowlist.internalerror', {
+          defaultMessage: 'Unable to create allowlist items',
+        }),
+      });
+    } else {
+      const cl = context.core.elasticsearch.dataClient.callAsCurrentUser;
+      hydrateWhitelistCache(cl);
+
+      // Whitelist IDs are generated by ES, add them to each rule before returning
+      let idx = 0;
+      createdItemIDs.forEach((id: string) => {
```

Comment:
You could just use a traditional `for` loop here, since you're having to manage your own index anyway.